cpp/src/arrow/filesystem/azurefs_test.cc

```diff
@@ -71,56 +72,113 @@ using ::testing::Not;
 using ::testing::NotNull;
 
 namespace Blobs = Azure::Storage::Blobs;
-namespace Files = Azure::Storage::Files;
+namespace Core = Azure::Core;
+namespace DataLake = Azure::Storage::Files::DataLake;
 
-auto const* kLoremIpsum = R"""(
-Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
-incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis
-nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
-Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu
-fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in
-culpa qui officia deserunt mollit anim id est laborum.
-)""";
+class BaseAzureEnv : public ::testing::Environment {
+ protected:
+  std::string account_name_;
+  std::string account_key_;
+
+  BaseAzureEnv(std::string account_name, std::string account_key)
+      : account_name_(std::move(account_name)), account_key_(std::move(account_key)) {}
 
-class AzuriteEnv : public ::testing::Environment {
  public:
-  AzuriteEnv() {
-    account_name_ = "devstoreaccount1";
-    account_key_ =
-        "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/"
-        "KBHBeksoGMGw==";
-    auto exe_path = bp::search_path("azurite");
-    if (exe_path.empty()) {
-      auto error = std::string("Could not find Azurite emulator.");
-      status_ = Status::Invalid(error);
-      return;
+  ~BaseAzureEnv() override = default;
```

Comment:
It's a, perhaps over-defensive, habit I have: declaring a dtor and marking it with `override` guarantees that the root of the class-hierarchy has a virtual destructor.

It's also a good habit if you care about binary size in the codebase (not the case here): because you can have `~Klass() override;` in the header and make the dtor non-inline.

Reply:
I didn't invent this rule. Someone explained it to me. It might be in the C++ Core Guidelines document which I haven't read, but I worked with people who did in the past.

Reply:
Hmm, I understand it if you're deriving an application-specific class, but you can pretty much count on `testing::Environment` having a virtual destructor. I don't understand the need to verify it explicitly.

Reply:
It's a "don't think, just do it" rule, but I will remove all the dtors here to cleanup.